R/Model-class.R

```diff
@@ -931,338 +931,224 @@ LogisticLogNormalMixture <- function(mean,
+    )
+  ),
+  contains = "Model",
+  validity = v_model_dual_endpoint
+)
 
-                     o$result()
-                 })
+## constructor ----
 
-##' Initialization function for the "DualEndpoint" class
-##'
-##' @param mu see \code{\linkS4class{DualEndpoint}}
-##' @param Sigma see \code{\linkS4class{DualEndpoint}}
-##' @param refDose see \code{\linkS4class{DualEndpoint}} (default: 1)
-##' @param useLogDose see \code{\linkS4class{DualEndpoint}}
-##' (default: \code{FALSE})
-##' @param sigma2W see \code{\linkS4class{DualEndpoint}}
-##' @param rho see \code{\linkS4class{DualEndpoint}}
-##' @return the \code{\linkS4class{DualEndpoint}} object
-##'
-##' @export
-##' @keywords methods
-DualEndpoint <- function(mu,
-                         Sigma,
-                         refDose=1,
-                         useLogDose=FALSE,
+#' @rdname DualEndpoint-class
+#'
+#' @param mean (`numeric`)\cr for the probit toxicity model, the prior mean vector.
+#' @param cov (`matrix`)\cr for the probit toxicity model, the prior covariance
+#'   matrix. The precision matrix is internally calculated as an inverse of `cov`.
+#' @param ref_dose (`number`)\cr for the probit toxicity model, the reference dose.
+#' @param use_log_dose (`flag`)\cr for the probit toxicity model, whether a log
+#'   transformation of the (standardized) dose should be used?
+#' @param sigma2W (`numeric`)\cr either a fixed value for the biomarker variance,
+#'   or a numeric vector with elements named `a` and `b` for the Inverse-Gamma
+#'   prior parameters.
+#' @param rho (`numeric`)\cr either a fixed value for the correlation
+#'   (between -1 and 1), or a named vector with elements named `a` and `b` for
+#'   the Beta prior on the transformation `kappa = (rho + 1) / 2`, which is in
+#'   (0, 1). For example, `a = 1, b = 1` leads to a uniform prior on `rho`.
+#'
+#' @export
+#'
+DualEndpoint <- function(mean,
+                         cov,
+                         ref_dose = 1,
+                         use_log_dose = FALSE,
                          sigma2W,
-                         rho)
-{
-    ## Find out which parameters are fixed
-    useFixed <- list()
-    for(parName in c("sigma2W", "rho"))
-    {
-        useFixed[[parName]] <-
-            identical(length(get(parName)), 1L)
-    }
-
-    ## build together the prior model and the parameters
-    ## to be saved during sampling
-    ## ----------
-
-    ## start with this:
-
-    modelspecs <-
-        list(mu=mu,
-             PrecBetaZ=solve(Sigma),
-             refDose=refDose## ,
-             ## low=c(-10000, 0),
-             ## high=c(0, 10000)
-             )
-
-    priormodel <-
-        function(){
-            ## priors for betaW: defined in subclasses
-
-            ## the bivariate normal prior for the
-            ## probit coefficients
-            log.betaZ[1:2] ~ dmnorm(mu[], PrecBetaZ[,])
+                         rho) {
+  use_fixed <- sapply(list(sigma2W = sigma2W, rho = rho), is.scalar)
```

Comment:
I would write this more explicit, since it is only 2 elements, e.g. 
```suggestion
  use_fixed <- list(sigma2W = is.scalar(sigma2W), rho = is.scalar(rho))
```